### PR TITLE
Make Solr page field sortable and use sorting in SearchInDocument plugin.

### LINKED ISSUE #671: Pagination of SearchInDocument results does not work
## Description

The search in document plugin lists 20 search results per page. Click on "next" shows the same results again.

## Reproduction

Example: http://digital.slub-dresden.de/id351357262

1. Search for "Pieschen". 
2. You get 20 results from page 3 to 73
3. Click "Next"
4. You get the same 20 results from page 3 to 73
5. The "Next"-button disappears

## Expected Behavior

Click to "next" should load the next 20 results.

There are also much more results between page 3 and 73 which are never shown!

## Screenshots and Examples

Digitale Sammlungen  Auf der Suche nach Zukunft  Das Beispiel Pieschen (2)

## Environment

 - TYPO3 version: 9.5.30
 - Kitodo.Presentation: 3.2.2

## Additional Context

The reason for this is the sorting done in JavaScript and not by Solr. It is not possible to sort by pages currently, because the Solr field "page" is not sortable.

### PR DESCRIPTION
**Pull request by albig**
This should fix #671.

### CODE REVIEW

**Comment by albig** [comment]
So far, this fixes #671 only partly.

Digitale Sammlungen  Auf der Suche nach Zukunft  Das Beispiel Pieschen (3)

- [x] Show the first 20 results from page 1 to 23
- [x] Click on "Next" shows still the same results
- [x] There is no "Back" button or another "Next" button

**Comment by albig** [comment]
Now the search should work as expected.

Please review @beatrycze-volk!

Digitale Sammlungen  Auf der Suche nach Zukunft  Das Beispiel Pieschen (4)

**Review by ByteParty — DISMISSED** [review]

**Review by beatrycze-volk — DISMISSED** [review]

File: Classes/Plugin/Eid/SearchInDocument.php [hunk 1]
```diff
@@ -59,7 +59,8 @@ public function main(ServerRequestInterface $request)
             $query = $solr->service->createSelect();
             $query->setFields([$fields['id'], $fields['uid'], $fields['page']]);
             $query->setQuery($fields['fulltext'] . ':(' . Solr::escapeQuery((string) $parameters['q']) . ') AND ' . $fields['uid'] . ':' . intval($parameters['uid']));
-            $query->setStart($count)->setRows(20);
+            $query->setStart($start)->setRows(20);
+            $query->addSort('page', $query::SORT_ASC);
```
Comment (beatrycze-volk) [inline Classes/Plugin/Eid/SearchInDocument.php]:
```suggestion
            $query->addSort($fields['page'], $query::SORT_ASC);
```

We should use configured field not string directly. There are possibilities that actually _page_ field is called differently.

Comment (albig) [reply Classes/Plugin/Eid/SearchInDocument.php]: Very good point. Thank you.

**Review by beatrycze-volk — APPROVED** [review]

**Pull request merged**